### ISSUE TITLE
ForbidCheckedExceptionInCallable error is reported in the wrong file

### ISSUE THREAD

**Issue by robotomarvin** [shipmonk-rnd/phpstan-rules#341]
When there is an error in the trait, that error gets reported on the right line, but on wrong file, in the examples you can find error in `FaultyTrait::failing`, that error gets wrongly reported in `ClassWithTrait.php` file 

I have made a reproducer: https://github.com/robotomarvin/forbid-checked-exception-in-callable-rule-error-reproducer.

I believe the problem is that the error is being returned on a node that have different scope (file in that scope) then the original node. 

## Inline version:

`ClassWithTrait.php`
```php
<?php

class ClassWithTrait
{
    use FaultyTrait;

    public function inClassTest()
    {
        // @phpstan-ignore-next-line shipmonk.checkedExceptionInCallable (this works properly)
        $this->acceptCallable($this->throw(...));
    }
}
```

`FaultyTrait.php`
```php
<?php

trait FaultyTrait
{
    /**
     * @throws \Exception
     */
    public function throw(): void
    {
        throw new \Exception();
    }

    public function acceptCallable(callable $c): void
    {

    }

    public function failing(): void
    {
        // @phpstan-ignore-next-line shipmonk.checkedExceptionInCallable (this is reported in `ClassWithTrait` file and this ignore is ignored)
        $this->acceptCallable(function() {
            $this->throw();
        });

    }
}
```
